packages/graphiql/src/components/DocExplorer/TypeLink.tsx

```diff
@@ -5,43 +5,45 @@
  *  LICENSE file in the root directory of this source tree.
  */
 
+import { useExplorerContext } from '@graphiql/react';
+import { GraphQLType, isListType, isNonNullType } from 'graphql';
 import React from 'react';
-import {
-  GraphQLList,
-  GraphQLNonNull,
-  GraphQLType,
-  GraphQLNamedType,
-} from 'graphql';
-import { OnClickTypeFunction } from './types';
-
-import { Maybe } from '../../types';
 
 type TypeLinkProps = {
-  type?: Maybe<GraphQLType>;
-  onClick?: OnClickTypeFunction;
+  type: GraphQLType;
 };
 
 export default function TypeLink(props: TypeLinkProps) {
-  const onClick = props.onClick ? props.onClick : () => null;
-  return renderType(props.type, onClick);
-}
+  const { push } = useExplorerContext({ nonNull: true, caller: TypeLink });
+
+  if (!props.type) {
+    return null;
+  }
 
-function renderType(type: Maybe<GraphQLType>, onClick: OnClickTypeFunction) {
-  if (type instanceof GraphQLNonNull) {
-    return <span>{renderType(type.ofType, onClick)}!</span>;
+  const type = props.type;
+  if (isNonNullType(type)) {
+    return (
+      <>
+        <TypeLink type={type.ofType} />!
+      </>
+    );
   }
-  if (type instanceof GraphQLList) {
-    return <span>[{renderType(type.ofType, onClick)}]</span>;
+  if (isListType(type)) {
+    return (
+      <>
+        [<TypeLink type={type.ofType} />]
+      </>
```

Comment:
no need for fragment wrapper
```suggestion
      <TypeLink type={type.ofType} />
```